weaviate/collection/collection.py

```diff
@@ -27,22 +31,73 @@
 from weaviate.util import _capitalize_first_letter
 
 
-class CollectionObject(CollectionObjectBase, Generic[Properties]):
+ITERATOR_CACHE_SIZE = 100
+
+FULL_METADATA_QUERY = MetadataQuery(
+    uuid=True,
+    creation_time_unix=True,
+    last_update_time_unix=True,
+    distance=True,
+    certainty=True,
```

Comment:
this could be a problem as certainty is not compatible with everything. Should we just have two arguments in the iterate() method? return_props and return_metadata just like in the query functions?

Reply:
Having both is a good idea. That will help with clarity around what the `return_properties` is doing there and its relation to the passed in (possibly generic) value